src/Voron/Data/Sets/SetLeafPage.cs

```diff
@@ -80,20 +80,21 @@ public void Init(long baseline)
             Header->NumberOfRawValues = 0;
         }
 
-        public struct Iterator : IDisposable
+        public struct Iterator 
         {
             private SetLeafPage _parent;
-
+            private const int MoveNextBufferSize = 32;
+            private int _moveNextIndex, _moveNextLength;
             private int _compressIndex, _compressLength;
             private CompressedHeader _compressedEntry;
             private long _lastVal;
             private int _rawValuesIndex, _compressedEntryIndex;
             private PForDecoder.DecoderState _decoderState;
             private bool _hasDecoder;
 
-            private ByteString _scratchMemory;
-            private ByteStringContext<ByteStringMemoryCache>.InternalScope _scratchScope; 
-            
+            private fixed int _pforBuffer[PForEncoder.BufferLen];
+            private fixed long _moveNextBuffer[MoveNextBufferSize];
```

Comment:
Need to check that this isn't causing problems, no?

Reply:
Also, need to set `SkipInit`

Reply:
@redknightlois ping